### PR TITLE
Support dispatching logical to physical experts

### PR DESCRIPTION
**Pull request by fzyzcjy**
## Motivation

subtract diff from https://github.com/sgl-project/sglang/pull/4957

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/managers/expert_location_dispatch.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    def init_new(cls, layer_id: int):
+        ep_dispatch_algorithm = global_server_args_dict["ep_dispatch_algorithm"]
+        expert_location_metadata = get_global_expert_location_metadata()
+
+        if ep_dispatch_algorithm is None:
+            return None
+
+        return cls(
+            ep_dispatch_algorithm=ep_dispatch_algorithm,
+            partial_logical_to_rank_dispatch_physical_map=expert_location_metadata.logical_to_rank_dispatch_physical_map[
```
Comment (yizhang2077) [inline python/sglang/srt/managers/expert_location_dispatch.py:45]: can not find `logical_to_rank_dispatch_physical_map` in `ExpertLocationMetadata`?

**Review by yizhang2077 — APPROVED** [review]

File: python/sglang/srt/managers/expert_location_dispatch.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import torch
+
+from sglang.srt.managers.expert_location import get_global_expert_location_metadata
+from sglang.srt.managers.schedule_batch import global_server_args_dict
+
+
+@dataclass
+class ExpertLocationDispatchInfo:
+    ep_dispatch_algorithm: Literal["static", "random"]
+    partial_logical_to_rank_dispatch_physical_map: torch.Tensor
```
Comment (yizhang2077) [inline python/sglang/srt/managers/expert_location_dispatch.py:28]: can we note these tensor shape?

Comment (fzyzcjy) [reply python/sglang/srt/managers/expert_location_dispatch.py:28]: sure
